torchdynamo/convert_frame.py

```diff
@@ -296,6 +296,11 @@ def debug_print(prefix):
                 print()
             assert output.guards is not None
             CleanupManager.instance[code] = output.cleanups
+
+            if "self" in frame.f_locals and istype(frame.f_locals["self"], torch.nn.ModuleList):
+                # Protected by module invalidation, see NN_MODULE
+                output.guards = []
```

Comment:
Won't this remove the call to `NNModuleChangeTrackerUtil.setup()` so nothing will be protected?

Reply:
Only for the module list, which should already be a property on the module. I can add a specific test for this though.

Reply:
Could this remove unrelated guards?  (For example globals or inputs?)

Reply:
Yea, maybe, let me try without this, its a bit more heavy handed than I'd like anyway.